lib/seam/request.rb

```diff
@@ -49,35 +49,51 @@ def on_complete(env)
           return if env.success?
 
           status_code = env.status
-          request_id = env.response_headers["seam-request-id"]
 
           retry_statuses = @retry_options.fetch(:retry_statuses, [])
           if retry_statuses.include?(status_code)
             raise Faraday::RetriableResponse.new(nil, env.response)
```

Comment:
I'm surprised we have to do this manually vs the retry middleware does this first before the Seam error handler.

Reply:
Oh yeah, I hadn't thought about this. I think we can achieve that by putting the retry middleware last in the Faraday connection builder, since the last middleware attached is the first to process the response. Nice catch!

Reply:
bbcf4e5062f4317f3c688c05ab93e837cd6d6c3d